ignite/cmd/scaffold.go

```diff
@@ -109,6 +113,51 @@ func scaffoldType(
 	return nil
 }
 
+func addGitChangesVerifier(cmd *cobra.Command) *cobra.Command {
+	cmd.Flags().AddFlagSet(flagSetYes())
+
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
What if in the future we will want to add additional PreRunLogic for one of these commands? This way it overrides all PreRunE hooks defined on the command.
I would consider to add this hook to all target commands directly in order to have an ability to extend PreRunE logic in the future.

Reply:
Yes that also what I remember. That's a bit weird side of Cobra, I don't know why they won't get executed by one by and instead they overwrite each other.

I really like this middleware like approach but if the overwrites still happen, I would say we could go with the way that Petr recommends to prevent accidental overwrites.

Later, we could refactor all the cobra command initializations by abstracting them in a separate package and manage hook ourselves to benefit from these middleware style.

Reply:
What do you think about:

preRunFun = cmd.PreRunE;
cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
if ( preRunFun != nil) preRunFun();

.... Check changes 
}

Reply:
Looks cool!